tests/macros/addonsidebar.test.js

```diff
@@ -0,0 +1,168 @@
+    ]),
+    'fr': new Set([
+        'Démarrage',
+        'Concepts',
+        'Interface Utilisateur',
+        'Mode d\'emploi',
+        'Portage',
+        'Déroulement avec Firefox',
+        'Les API JavaScript',
+        'Clés de manifeste',
+        'Thème de navigateur',
+        'Publication de votre extension',
+        'Distribuer votre module',
+        'Canaux de discussions'
+    ]),
+    'ja': new Set([
+        '始めましょう',
+        '概念',
+        'ユーザーインターフェイス',
+        '逆引きリファレンス',
+        '移行',
+        'Firefox でのワークフロー',
+        'JavaScript API 群',
+        'Manifest keys',
+        'ブラウザのテーマ',
+        'アドオンを公開する',
+        'アドオンの配布',
+        'チャンネル'
+    ]),
+};
+
+const MANIFEST_SLUG = 'Mozilla/Add-ons/WebExtensions/manifest.json';
+
+function getMockResultForFetchJSONResource(doc_url) {
+    const locale = url.parse(doc_url).pathname.split('/')[1];
+    return {
+        locale: `${locale}`,
+        url: `/${locale}/docs/${MANIFEST_SLUG}`,
+        subpages: [
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/author`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/author`,
+                title: 'author'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/background`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/background`,
+                title: 'background'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/theme`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/theme`,
+                title: 'theme'
+            },
+            {
+                locale: `${locale}`,
+                url: `/${locale}/docs/${MANIFEST_SLUG}/version`,
+                subpages: [],
+                slug: `${MANIFEST_SLUG}/version`,
+                title: 'version'
+            },
+        ],
+        slug: MANIFEST_SLUG,
+        title: 'manifest.json'
+    };
+}
+
+function checkSidebarDom(html, locale, is_under_web_ext_api=false) {
+    // Lint the HTML
+    expect(lintHTML(html)).toBeFalsy();
+    const dom = JSDOM.fragment(html);
+    const section = dom.querySelector('section.Quick_links');
+    // Check the basics
+    expect(section).toBeTruthy();
+    // Check the total number of top-level list items that can be toggled
+    expect(
+        section.querySelectorAll('ol > li.toggle').length
+    ).toEqual(SUMMARIES[locale].size);
+    // Check that all links reference the proper locale or use https
+    const num_total_links = section.querySelectorAll('a[href]').length;
+    const num_valid_links = section.querySelectorAll(
+        `a[href^="/${locale}/docs/Mozilla/Add-ons"], a[href^="https://"]`
+    ).length;
+    expect(num_valid_links).toEqual(num_total_links);
+    // Check that one of the list item's details are open by default (or not)
```

Comment:
As far as I can tell, this bit is testing specifically that if the slug is under 'WebExtensions/API' then the 'JavaScript APIs' summary is open, and that otherwise the summaries are all closed.

If this is right then I think it would be clearer to say that in this comment (and also perhaps in the test, to test that the open summary is in fact  'JavaScript APIs').